src/App.tsx

```diff
@@ -1,50 +1,108 @@
-import React from 'react';
+import React, { useState } from 'react';
 import 'bulma/css/bulma.css';
 import './App.scss';
+import cn from 'classnames';
+
+type Props = {
+  initialGoods: string[];
+};
+
+enum SortType {
+  Alphabetical = 'Alphabetical',
+  ByLength = 'ByLength',
+  None = 'None',
+}
+
+export const App: React.FC<Props> = ({ initialGoods }) => {
+  const [goods, setGoods] = useState([...initialGoods]);
+  const [sortType, setSortType] = useState<SortType>(SortType.None);
+  const [isReversedActive, setIsReversedActive] = useState<boolean>(false);
+
+  function sortGoods(type: SortType) {
+    let sortedGoods = [...goods];
+
+    switch (type) {
+      case SortType.Alphabetical:
+        sortedGoods.sort((one, two) => one.localeCompare(two));
+        setIsReversedActive(false);
+        break;
+      case SortType.ByLength:
+        sortedGoods.sort((one, two) => one.length - two.length);
+        setIsReversedActive(false);
+        break;
+      case SortType.None:
+        sortedGoods = [...initialGoods];
+        setIsReversedActive(false);
+        break;
```

Comment:
just create some getPreparedGoods function that will sort and reverse our array and return the result

```
const preparedGoods = getPreparedGoods(goods, sortType, isReversed)
```
and then use preparedGoods everywhere